src/_data/Auth.js

```diff
@@ -1,56 +1,52 @@
-import { rehydratedStorePromise, store } from '../_store/configureStore';
+import { store } from '../_store/configureStore';
 import * as LiveData from './LiveData';
 import { updateToken, signinFieldUpdate, updateAppInfo } from '../_actions';
 import { trackUserId } from '../_utils/Analytics';
 
-let isAuthorized = false;
+export const tryAuth = (st) => {
+    const newState = st.getState();
+    if (!newState.account) {
+        return Promise.reject();
+    }
+
+    const token = newState.account.get('token');
+    if (!token) {
+        st.dispatch(signinFieldUpdate('progress', false));
+        st.dispatch(signinFieldUpdate('tokenNotEntered', true));
+        return Promise.reject();
+    }
+    st.dispatch(updateAppInfo('authorized', false));
+    return LiveData.api.authorize(token).then(
+        response => {
+            st.dispatch(signinFieldUpdate('credentialsInvalid', false));
```

Comment:
Let's use store everywhere instead of 'st' :)
Also you can use bindActionCreators(AllActions, store.dispatch) to then call them by name and not dispatch, seems a bit cleaner.